src/python/WMComponent/DBS3Buffer/MySQL/NewSubscription.py

```diff
@@ -40,15 +40,17 @@ def _createPhEDExSubBinds(self, datasetID, subscriptionInfo, custodialFlag):
 
         binds = []
         for site in sites:
-            binds.append({'id': datasetID,
-                          'site': site,
-                          'custodial': custodialFlag,
-                          'auto_approve': 1 if site in subscriptionInfo['AutoApproveSites'] else 0,
-                          'move': isMove,
-                          'priority': subscriptionInfo['Priority'],
-                          'phedex_group': phedex_group,
-                          'delete_blocks': delete_blocks,
-                          'dataset_lifetime': subscriptionInfo['DatasetLifetime']})
+            bind = {'id': datasetID,
+                    'site': site,
+                    'custodial': 0 if custodialFlag is None else custodialFlag,
+                    'auto_approve': 1 if site in subscriptionInfo['AutoApproveSites'] else 0,
+                    'move': 0 if isMove is None else isMove,
+                    'priority': subscriptionInfo['Priority'],
```

Comment:
can you make proper default here, e.g. `subscriptionInfo.get('Priority', "some-default-value")`. And, the value should be validated too to satisfy allowed values.

Reply:
Request priority default comes from the upper spec file, here (and data type is also validated in there):
https://github.com/dmwm/WMCore/blob/master/src/python/WMCore/WMSpec/StdSpecs/StdBase.py#L1025

Having default values defined in multiple places is just source for a future error.